### PR TITLE
Fix logic error in check_integrity

### PR DESCRIPTION
**Pull request by pmeier**
See #869

### CODE REVIEW

File: torchvision/datasets/utils.py [hunk 1]
```diff
@@ -17,20 +17,26 @@ def bar_update(count, block_size, total_size):
...
-    md5o = hashlib.md5()
-    with open(fpath, 'rb') as f:
-        # read in 1MB chunks
-        for chunk in iter(lambda: f.read(1024 * 1024), b''):
-            md5o.update(chunk)
-    md5c = md5o.hexdigest()
-    if md5c != md5:
-        return False
-    return True
+    else:
```
Comment (fmassa) [inline torchvision/datasets/utils.py]: This else block is not required I believe, and you can remove one level of indentation

File: torchvision/datasets/utils.py [hunk 2]
```diff
@@ -17,20 +17,26 @@ def bar_update(count, block_size, total_size):
     return bar_update
 
 
+def calc_md5(fpath, chunk_size=1024*1024):
```
Comment (fmassa) [inline torchvision/datasets/utils.py]: maybe a slightly more descriptive name, like `calculate_md5`?

**Comment by pmeier** [comment]
> Also, can you add some tests in `test_datasets.py` for both `check_integrity` and `check_md5`?

Sure. Should I push them to this branch or open another PR?

**Comment by soumith** [comment]
thanks!

**Comment by fmassa** [comment]
@pmeier just send a new PR

**Pull request merged**